File: template/ios/AppSplash/AppSplashModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_REMAP_MODULE(AppSplash, AppSplashModule, NSObject)

RCT_EXTERN_METHOD(hide:(BOOL)fade
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(isVisible:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: template/ios/AppSplash/AppSplashModule.swift
import React

@objc(AppSplashModule)
final class AppSplashModule: NSObject {

  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc func hide(
    _ fade: Bool,
    resolve: @escaping RCTPromiseResolveBlock,
    reject _: @escaping RCTPromiseRejectBlock,
  ) {
    AppSplash.hide(fade: fade) { resolve(nil) }
  }

  @objc func isVisible(
    _ resolve: @escaping RCTPromiseResolveBlock,
    reject _: @escaping RCTPromiseRejectBlock,
  ) {
    DispatchQueue.main.async { resolve(AppSplash.isVisible) }
  }
}
